src/parser/key.rs

```diff
@@ -17,11 +23,48 @@ parse!(unquoted_key() -> &'a str, {
     take_while1(is_unquoted_char)
 });
 
-// key = unquoted-key / basic-string / literal-string
-parse!(key() -> (&'a str, InternalString), {
-    recognize_with_value(choice((
-        basic_string(),
-        literal_string().map(|s: &'a str| s.into()),
-        unquoted_key().map(|s: &'a str| s.into()),
-    )))
+// From spec:
+// simple-key = quoted-key / unquoted-key
+// actual: key = unquoted-key / basic-string / literal-string
```

Comment:
it was implemented using an older version of spec, so if it has changed, the comment is worth updating as well

Reply:
You mean the code should be updated? I added the comment to remind us the parser branches differently than the spec but the result is the same.

Reply:
I've adapted the spec in comments to the parser in some places, so sometimes it's not a direct copy-paste. From your comment change it's not clear whether they are equivalent, so it would be good to update the comment to clarify that.